deepspeed/runtime/zero/stage2.py

```diff
@@ -300,7 +306,8 @@ def __init__(self,
             # each process will compute on a different part of the partition
             data_parallel_partitions = self.get_data_parallel_partitions(
                 self.fp16_groups_flat[i],
-                i)
+                i,
+                self.nccl_start_alignment_factor)
```

Comment:
@amathews-amd since you have already aligned self.fp16_groups_flat, do we need this alignment on 310 again? We should be able to leave the get_data_parallel_parititons method unmodified  no?

Reply:
flatten_dense_tensors_aligned() is not a good name (part of the original code). All it actually does is padding at the end.

Reply:
The padding at the end is enough to keep things aligned

Reply:
Thanks. I agree its not fully explicit but its not completely implicit either since you are making the alignment when creating the buffer. I think we can also add explicit assertions once we get the partitions from get_data_parallel_parititons  to ensure directly in Stage 2 that the alignment is satisfied instead in addition to having the unit tests. Since the assertion is only in initialization, this will only be done once at the beginning.

Reply:
Done.

Reply:
I have made the modifications as recommended.

Reply:
Thank you. Looks great to me..